src/android/StatusBar.java

```diff
@@ -164,4 +181,21 @@ private void setStatusBarBackgroundColor(final String colorPref) {
             }
         }
     }
+
+    private void setStatusBarTransparent(final boolean transparent) {
+        if (Build.VERSION.SDK_INT >= 21) {
+            final Window window = cordova.getActivity().getWindow();
+            if (transparent) {
+                window.getDecorView().setSystemUiVisibility(
+                        View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
+                window.setStatusBarColor(Color.TRANSPARENT);
```

Comment:
Reflection is not the approach used by Android Studio in handling any of the platform specific code. In fact, reflection has other drawbacks such as being slow and making debugging harder.

The code uses a version check which is recognized by the standard Android build process and it works fine also on e.g. Android API 19 and other versions.